### PR TITLE
convert ZMSyncStrategy.callingRequestStrategy to swift

### PR DESCRIPTION
**Pull request by billypchan**
## What's new in this PR?

Prepare for converting `CallingRequestStrategy` to non-objc class, convert this method to Swift.

### CODE REVIEW

File: Tests/Source/Integration/APNSTests.m [hunk 1]
```diff
@@ -31,6 +31,7 @@ @implementation APNSTests
 
 - (void)testThatItFetchesTheNotificationStreamWhenReceivingNotificationOfTypeNotice
 {
+    ///TODO: crash on M1?
```
Comment (KaterinaWire) [inline Tests/Source/Integration/APNSTests.m:34]: What does this comment mean?

**Comment by KaterinaWire** [comment]
@billypchan Why did you decide to convert `CallingRequestStrategy` to non-objc class? Is it a part of some refactoring?

File: Tests/Source/Integration/APNSTests.m (hunk 1 above)
Comment (billypchan) [inline Tests/Source/Integration/APNSTests.m:34]:
```suggestion
```

Comment (billypchan) [reply Tests/Source/Integration/APNSTests.m]: cleared

**Comment by billypchan** [comment]
@KaterinaWire since `CallingRequestStrategy` is required to be non-objc in XCFramework. I will open another PR to covert all the required class to non-objc.

**Review by KaterinaWire — APPROVED** [review]

**Review by David-Henner — APPROVED** [review]

**Comment by billypchan** [comment]
@zenkins release patch

**Pull request merged**